keras_cv/layers/preprocessing/random_translation.py

```diff
@@ -0,0 +1,256 @@
+                "RandomTranslation received both `x_factor=None` and `y_factor=None`.  "
+                "As a result, the layer will perform no augmentation."
+            )
+        self.interpolation = interpolation
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.seed = seed
+        self.bounding_box_format = bounding_box_format
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_image(self, image, transformation, **kwargs):
+        return self._translate_image(image, transformation)
+
+    def get_random_transformation(self, **kwargs):
+        translate_horizontal = 0.0
+        translate_vertical = 0.0
+        if self.x_factor:
+            translate_horizontal = self.x_factor()
+        if self.y_factor:
+            translate_vertical = self.y_factor()
+        return {
+            "translate_horizontal": tf.cast(translate_horizontal, dtype=tf.float32),
+            "translate_vertical": tf.cast(translate_vertical, dtype=tf.float32),
+        }
+
+    def _translate_image(self, image, transformation):
+        image = preprocessing.ensure_tensor(image, self.compute_dtype)
+        image_shape = image.shape
+        image_width = tf.cast(image_shape[1], tf.float32)
+        image_height = tf.cast(image_shape[0], tf.float32)
+        element = tf.stack(
+            [
+                transformation["translate_horizontal"] * image_width,
+                transformation["translate_vertical"] * image_height,
+            ],
+        )
+        output = preprocessing.transform(
+            tf.expand_dims(image, 0),
+            preprocessing.get_translation_matrix(tf.expand_dims(element, axis=0)),
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            interpolation=self.interpolation,
+        )
+        output = tf.squeeze(output, 0)
+        output.set_shape(image_shape)
+        return output
+
+    @staticmethod
+    def _translate_bounding_boxes_horizontal(bounding_boxes, dx):
+        x1, x2, x3, x4, rest = tf.split(
+            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
+        )
+        output = tf.stack(
+            [
+                x1 + dx,
+                x2,
+                x3 + dx,
+                x4,
+                rest,
+            ],
+            axis=-1,
+        )
+        output = tf.squeeze(output, axis=1)
+        return output
+
+    @staticmethod
+    def _translate_bounding_boxes_vertical(bounding_boxes, dy):
+        x1, x2, x3, x4, rest = tf.split(
+            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
+        )
+        output = tf.stack(
+            [
+                x1,
+                x2 + dy,
+                x3,
+                x4 + dy,
+                rest,
+            ],
+            axis=-1,
+        )
+        output = tf.squeeze(output, axis=1)
+        return output
+
+    def augment_bounding_boxes(
```

Comment:
As implemented, this works as-intended for `fill_mode="constant"` (and I think "nearest"), but not for "reflect" and "wrap".

When wrapping or reflecting, we'll need to create new bounding boxes (or expand existing ones) for the reflected or wrapped pixels that were previously inside of a bounding box (and then potentially prune the new ones if they are unreasonably small).

Here's an example I produced from running your example script:
<img width="231" alt="Screen Shot 2022-09-22 at 10 54 26 AM" src="https://user-images.githubusercontent.com/3072903/191811765-fbc80f27-4a27-4b8f-9adb-6ad3b6a7233e.png">

Notice that the mirrored cow on the right does not have a bounding box on it (but it should).

If you'd like to, for now we can raise an exception if this layer is called on bounding boxes with one of the non-constant fill modes, and leave this code as-is. Otherwise, let's update it to support the other fill modes.

Reply:
I tried to check how other frameworks handle that (ie albumentations) and I don't think they create new bounding boxes. There's also a case, when reflected box' class might no longer be a valid. Anyway I think you're right and it should be added.

Reply:
we should raise an error if you’re augmenting bounding boxes with reflect or wrap fill mode until we implement a projective transform for bounding boxes IMO